tests/jobs/test_backfill_job.py

```diff
@@ -350,9 +377,8 @@ def test_backfill_respect_max_active_tis_per_dag_limit(self, mock_log, dag_maker
         )
         dag_maker.create_dagrun(state=None)
 
-        executor = MockExecutor()
-
-        job = Job(executor=executor)
+        job = Job()
+        executor = job.executor
```

Comment:
Nit:  (Here and elsewhere) With the changes you made, it seems a bit unnecessary to set `executor = job.executor`.  Perhaps consider just inlining that.  You only use it twice in the test and it's not really clarifying or saving repetition much.

Reply:
Personally I liked that doing it this way minimized the number of lines of the diff (the rest of the code in this test can remain the same) and it is used twice so a variable seemed applicable